spec/griddler/cloudmailin/adapter_spec.rb

```diff
@@ -1,43 +1,120 @@
+      },
+      plain: 'hi',
+      headers: {
+        From: 'there@example.com',
+        To: 'Hello World <hi@example.com>',
+        Cc: 'emily@example.com'
+      }
+    )
+  end
 
   it 'normalizes parameters' do
-    expect(Griddler::Cloudmailin::Adapter.normalize_params(default_params)).to be_normalized_to({
+    expect(normalized).to be_normalized_to(
       to: ['Some Identifier <some-identifier@example.com>'],
       cc: ['emily@example.com'],
       from: 'Joe User <joeuser@example.com>',
       subject: 'Re: [ThisApp] That thing',
       text: /Dear bob/
-    })
+    )
   end
 
-  it 'passes the received array of files' do
-    params = default_params.merge({ attachments: [upload_1, upload_2] })
+  it 'normalizes parameters where the recipient has been BCCed' do
+    expect(normalized(bcc_params)).to be_normalized_to(
+      to: ['Some Identifier <some-identifier@example.com>'],
+      cc: ['emily@example.com'],
+      bcc: ['sandra@example.com'],
+      from: 'Joe User <joeuser@example.com>',
+      subject: 'Re: [ThisApp] That thing',
+      text: /Dear bob/
+    )
+  end
 
-    normalized_params = Griddler::Cloudmailin::Adapter.normalize_params(params)
-    expect(normalized_params[:attachments]).to eq [upload_1, upload_2]
+  it 'normalizes parameters where the recipient has been CCed' do
+    expect(normalized(cc_params)).to be_normalized_to(
+      cc: ['Some Identifier <some-identifier@example.com>'],
+      to: ['emily@example.com'],
+      from: 'Joe User <joeuser@example.com>',
+      subject: 'Re: [ThisApp] That thing',
+      text: /Dear bob/
+    )
+  end
+
+  it 'changes attachments to an array of files' do
+    params = default_params.merge(attachments: { '0' => upload_1, '1' => upload_2 })
+
+    normalized_params = normalized(params)
+    expect(normalized_params[:attachments]).to include(upload_1, upload_2)
+    expect(normalized_params[:attachments]).not_to include('0')
+    expect(normalized_params[:attachments]).not_to include('1')
   end
 
   it 'has no attachments' do
-    params = default_params
+    expect(normalized[:attachments]).to be_empty
+  end
+
+  it 'wraps to in an array' do
+    expect(normalized[:to]).to eq([default_params[:headers][:To]])
+  end
+
+  it 'wraps cc in an array' do
+    expect(normalized[:cc]).to eq([default_params[:headers][:Cc]])
+  end
 
-    normalized_params = Griddler::Cloudmailin::Adapter.normalize_params(params)
-    expect(normalized_params[:attachments]).to be_empty
+  it 'returns an array even if cc is empty' do
+    expect(normalized(nocc_params)[:cc]).to eq([])
+  end
+
+  def normalized(params = default_params)
+    Griddler::Cloudmailin::Adapter.normalize_params(params)
   end
 
   def default_params
     {
-      envelope: { to: 'Some Identifier <some-identifier@example.com>', from: 'Joe User <joeuser@example.com>' },
-      headers: { Subject: 'Re: [ThisApp] That thing', Cc: 'emily@example.com' },
+      envelope: {
+        to: 'some-identifier@example.com',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.